src/concat.js

```diff
@@ -13,8 +13,8 @@ var invoker = require('./invoker');
  * @category List
  * @sig [a] -> [a] -> [a]
  * @sig String -> String -> String
- * @param {Array|String} a
- * @param {Array|String} b
+ * @param {Array|String|Object} a
+ * @param {Array|String|Object} b
```

Comment:
I'd suggest using `{concat: function}` for both params to indicate that the types of each param should be the same (even if `concat` isn't actually used in the second param). Furthermore, the return type should be the same as both the params for `concat` too.

That particular test could be updated to something like the following to remove some of the ambiguity:
```js
function a(x) {
  return {
    x: x,
    concat: that => a(x + ' ' + that.x)
  };
}
eq(R.concat(a('z1'), a('z2')).x, 'z1 z2');
```